test/test_euler.py

```diff
@@ -0,0 +1,770 @@
+
+    p0 = 1.0
+
+    # === this next block tests 1,2,3 dimensions,
+    # with single and multiple nodes/states. The
+    # purpose of this block is to ensure that when
+    # all components of V = 0, the flux recovers
+    # the expected values (and p0 within tolerance)
+    # === with V = 0, fixed P = p0
+    tolerance = 1e-15
+    for ntestnodes in [1, 10]:
+        fake_dis = MyDiscr(dim)
+        mass = cl.clrandom.rand(
+            queue, (ntestnodes,), dtype=np.float64
+        )
+        energy = cl.clrandom.rand(
+            queue, (ntestnodes,), dtype=np.float64
+        )
+        mom = make_obj_array(
+            [
+                cl.clrandom.rand(
+                    queue, (ntestnodes,), dtype=np.float64
+                )
+                for i in range(dim)
+            ]
+        )
+        p = cl.clrandom.rand(queue, (ntestnodes,), dtype=np.float64)
+
+        for i in range(ntestnodes):
+            mass[i] = 1.0 + i
+            p[i] = p0
+            for j in range(dim):
+                mom[j][i] = 0.0 * mass[i]
+        energy = p / 0.4 + 0.5 * np.dot(mom, mom) / mass
+        q = flat_obj_array(mass, energy, mom)
+        p = eos.pressure(q)
+        flux = _inviscid_flux(fake_dis, q, eos)
+
+        logger.info(f"{dim}d flux = {flux}")
+
+        # for velocity zero, these components should be == zero
+        for i in range(2 * dim):
+            for j in range(ntestnodes):
+                assert flux[i][j].get() == 0.0
+
+        # The momentum diagonal should be p
+        # Off-diagonal should be identically 0
+        for i in range(dim):
+            for j in range(dim):
+                print(f"(i,j) = ({i},{j})")
+                if i != j:
+                    for n in range(ntestnodes):
+                        assert (
+                            flux[(2 + i) * dim + j][n].get() == 0.0
+                        )
+                else:
+                    for n in range(ntestnodes):
+                        assert (
+                            flux[(2 + i) * dim + j][n].get() == p[n]
+                        )
+                        assert (
+                            np.abs(
+                                flux[(2 + i) * dim + j][n].get()
+                                - p0
+                            ) < tolerance
+                        )
+
+
+@pytest.mark.parametrize("dim", [1, 2, 3])
+def test_inviscid_mom_flux_components(ctx_factory, dim):
+    r"""Constant pressure, V != 0:
+    Checks that the flux terms are returned in the proper
+    order by running only 1 non-zero velocity component at-a-time.
+    """
+
+    cl_ctx = ctx_factory()
+    queue = cl.CommandQueue(cl_ctx)
+    logger = logging.getLogger(__name__)
+
+    eos = IdealSingleGas()
+
+    class MyDiscr:
+        def __init__(self, dim=1):
+            self.dim = dim
+
+    p0 = 1.0
+
+    from mirgecom.euler import _inviscid_flux
+
+    tolerance = 1e-15
+    for livedim in range(dim):
+        for ntestnodes in [1, 10]:
```

Comment:
Do we gain much by testing this on both 1 and 10 nodes? Shouldn't one case of 10 suffice?

Reply:
I gained a lot of understanding about how the returned arrays were arranged by testing it both on 1 and 10.